### ISSUE TITLE
tessdata "fast" vs "best"

### ISSUE THREAD

**Issue by cneud** [OCR-D/ocrd_all#52]
The current value for ``TESSDATA_URL`` is ``https://github.com/tesseract-ocr/tessdata_fast``, in the context of OCR-D I propose to change the default to ``https://github.com/tesseract-ocr/tessdata_best`` instead.

**Comment by stweil** [comment]
Why do you think that would help? My experience is that tessdata_best is not significantly better (if it is better at all), but takes significantly more time.

The default for Linux distributions is tessdata_fast.

I think that in the context of OCR-D the models from tessdata* are not adequate because of their known bugs. So it is sufficient to get the eng, equ and osd models to satisfy Tesseract, but no other of the standard models will be needed.

The models from tessdata_best are useful as a starting point for training of new models.

**Comment by cneud** [comment]
>Why do you think that would help? My experience is that tessdata_best is not significantly better (if it is better at all), but takes significantly more time for processing a page.

My point was that now that we recommend to use ``ocrd_all`` as the basis to setup/deploy OCR-D in libraries, this is what libraries are going to use. In that context, I would argue that quality of the results is even more important than processing time and in my personal experience so far, I did actually encounter better results from the "tessdata_best" models in various cases. Unfortunately though I have never evaluated this properly and on a larger scale - do you know if anyone ever performed a direct comparison of the "best" vs "fast" models?

> I think that in the context of OCR-D the models from tessdata* are not adequate because of their known bugs. So it is sufficient to get the eng, equ and osd models to satisfy Tesseract, but no other of the standard models will be needed.

Then perhaps it would make sense to include your current best ``GT4HistOCR`` model directly in ``ocrd_all``?

> Ideally we should only have models for OCR-D which can be reproduced by a well defined training process.

Fully agreed, but I think we also should try to ensure sure that users who install ``ocrd_all`` ideally have a suitable model for German historic prints right away.

**Comment by stweil** [comment]
The current OCR-D ressembles a collection of building blocks, but plans what can be built are rare:

- I still don't know which OCR-D workflow works best and is recommended for typical prints.
- A lot of tools require models - not only all recognizer moduls, but also tools for segmentation and post-processing. In most cases these models are not part of the Git repositories and currently not installed by ocrd_all.

I think this is a scenario which only (or not even?) experts can handle, and it shows that more has to be done until OCR-D is ready to be used by everybody.

We only recently learned that it seems to be possible to use a single model for most historic prints, not only German ones (we already used it in Mannheim for French and Latin texts, too). Up to now, we use one of the models which were produced, without having evaluated all other new models. Every few weeks, the training process is still producing a new model. And I already know what should be made better next time when we start a new training.

Therefore I hesitate to add that model to ocrd_all now although I can understand you very well. I had written some [documentation](https://wiki.de.dariah.eu/pages/viewpage.action?pageId=108140929) for the OCR-D pilot libraries.

**Comment by cneud** [comment]
Agree with all your points. Nevertheless, since the expectation now is to install ``ocrd_all`` and get decent OCR results out-of-the-box, deploying good models and/or having clear documentation about the various models available and what their status is becomes paramount. I had also already started a document like this earlier, but yours is much more complete. I hope we can include this with the revamp of the website/docs @kba?

**Comment by kba** [comment]
The info on available models is really helpful, thanks @stweil We'll integrate that info into the documentation.

**Comment by bertsky** [comment]
Generally I second the idea that ocrd_all should aim to provide the best available tools and models at any given time, while making minimal assumptions on use-cases. 

Thus, since the Tesseract models trained by Google are still the best there is for contemporary text, and are a good and well-known reference point for historic text, I think they should always be included. (And as for `best` vs `fast`, perhaps we could again add a variable for user preference.)

>     * I still don't know which OCR-D workflow works best and is recommended for typical prints.

We already include some good example configurations in bertsky/workflow-configuration. I have been testing various workflows with existing processors and parameters on the OCR-D textual GT, and will continue to to update them to the best of my knowledge for best overall results. Of course, everyone is welcome to contribute results and different configurations.

>     * A lot of tools require models - not only all recognizer moduls, but also tools for segmentation and post-processing. In most cases these models are not part of the Git repositories and currently not installed by ocrd_all.

True, but we could easily make that happen here by adding respective makefile rules which checkout subrepos or download files from the web. This would fit in naturally into the current spirit of encapsulating knowledge about each module in a few standardized variable and rule definition lines. One target (say `models` would install them, another (say `model-info`) would describe what is already installed.

**Comment by bertsky** [comment]
Generally I second the idea that ocrd_all should aim to provide the best available tools and models at any given time, while making minimal assumptions on use-cases. 

Thus, since the Tesseract models trained by Google are still the best there is for contemporary text, and are a good and well-known reference point for historic text, I think they should always be included. (And as for `best` vs `fast`, perhaps we could again add a variable for user preference.)

>     * I still don't know which OCR-D workflow works best and is recommended for typical prints.

We already include some good example configurations in bertsky/workflow-configuration. I have been testing various workflows with existing processors and parameters on the OCR-D textual GT, and will continue to to update them to the best of my knowledge for best overall results. Of course, everyone is welcome to contribute results and different configurations.

>     * A lot of tools require models - not only all recognizer moduls, but also tools for segmentation and post-processing. In most cases these models are not part of the Git repositories and currently not installed by ocrd_all.

True, but we could easily make that happen here by adding respective makefile rules which checkout subrepos or download files from the web. This would fit in naturally into the current spirit of encapsulating knowledge about each module in a few standardized variable and rule definition lines. One target (say `models` would install them, another (say `model-info`) would describe what is already installed.


> I had written some [documentation](https://wiki.de.dariah.eu/pages/viewpage.action?pageId=108140929) for the OCR-D pilot libraries.

@stweil can you please increase the accessibility/visibity of that page for the other OCR-D participants in DARIAH?

**Comment by stweil** [comment]
I have not set any restrictions, so everybody can even enhance it. The existing restrictions are inherited, and I have no rights to change them.

**Comment by stweil** [comment]
> We already include some good example configurations in bertsky/workflow-configuration.

Perhaps such information and some examples could be added to README.md (which is currently rather technical, going deep into details of options for `make`).

**Comment by bertsky** [comment]
> I have not set any restrictions, so everybody can even enhance it. The existing restrictions are inherited, and I have no rights to change them.

Can anyone help? (I am logged into Confluence with my username `RobertSachunsky`. Perhaps there are some group separation rules in place?)

> Perhaps such information and some examples could be added to README.md (which is currently rather technical, going deep into details of options for `make`).

I could, but isn't that for the individual module's documentation alone? We could perhaps link to the submodule READMEs from the ocrd_all README (but in the past, at least on Github Markdown, these links never resolved). Or we encourage every module to use GH pages and link to that – in this case https://bertsky.github.io/workflow-configuration/

**Comment by stweil** [comment]
@EEngl52, can you help @bertsky?

**Comment by EEngl52** [comment]
@bertsky I just gave you access to the pilot library section in the Dariah wiki. We are thinking about using it in the relaunch of our old website as part of the description on the OCR-D-processors and this way making it more visible.

**Comment by bertsky** [comment]
@EEngl52 it worked – thanks!
